syntax = "proto3";

// Get protoc here https://github.com/protocolbuffers/protobuf/releases
// .\protoc --python_out=. --pyi_out=. Autogen.proto

message DataMap {
    map<string, string> data = 1;
}

message ReceiveReq {
    oneof Type {
        DataMap data_map = 1;
        string data = 2;
    }
    optional string sender = 3;
    optional bool request_reply = 4;
    optional bool silent = 5;
}

message Terminate {
    optional string sender = 1;
}

message GenReplyReq {
}

message GenReplyResp {
    optional string data = 1;
}

message PrepChat {
    string recipient = 1;
    bool clear_history = 2;
    bool prepare_recipient = 3;
}
